test/Examples/ContraSpec.hs

```diff
@@ -0,0 +1,84 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE GADTs #-}
+
+module Examples.ContraSpec where
+
+import           Test.Hspec
+
+import           Generics.Eot
+
+import           Data.Void
+import           Data.Functor.Contravariant
+import           Data.Functor.Contravariant.Divisible
+
+
+newtype Serializer a = Serializer { runSerializer :: a -> String }
+
+string :: Serializer String
+string = Serializer id
+
+int :: Serializer Int
+int = Serializer show
+
+double :: Serializer Double
+double = Serializer show
+
+bool :: Serializer Bool
+bool = Serializer show
+
+instance Contravariant Serializer where
+  contramap f (Serializer g) = Serializer (g . f)
+
+instance Divisible Serializer where
+  conquer = Serializer (const mempty)
+
+  divide toBC (Serializer sb) (Serializer sc) = Serializer $ \a ->
+    case toBC a of
+      (b, c) ->
+        let bBytes = sb b
+            cBytes = sc c
+        in bBytes ++ cBytes
+
+instance Decidable Serializer where
+  lose f = Serializer $ \a -> absurd (f a)
+  choose split l r = Serializer $ \a ->
+    either (runSerializer l) (runSerializer r) (split a)
+
+infixr 4 >*<
+(>*<) :: Divisible f => f a -> f b -> f (a, b)
+(>*<) = divided
+
+infixr 3 >|<
+(>|<) :: Decidable f => f a -> f b -> f (Either a b)
+(>|<) = chosen
+
+contraSum :: (HasEot a, Contravariant f) => f (Eot a) -> f a
+contraSum = contramap toEot
+
+contraProduct :: (HasEot a, Decidable f, Eot a ~ Either b Generics.Eot.Void) => f b -> f a
+contraProduct = contramap toEot . flip chosen lost
+
+data Identifier = StringId String | IntId Int
+  deriving (Eq, Ord, Show, Generic)
+
+identifier :: Serializer Identifier
+identifier = contraSum $
+  string >*< conquer >|<
+  int    >*< conquer >|<
+  lost
+
+data Blob = Blob Int Identifier Bool
+  deriving (Eq, Ord, Show, Generic)
+
+blob :: Serializer Blob
+blob = contraProduct $
+  int        >*<
+  identifier >*<
+  bool       >*<
+  conquer
+
+spec :: Spec
+spec = do
+  describe "contravariant" $ do
+    it "allows you to use Decidable with Void" $ do
+      runSerializer blob (Blob 1 (StringId "hi") False) `shouldBe` "1hiFalse"
```

Comment:
It seems that a lot of the code in this module is not necessary for a minimal test case. For example the type `Blob` seems unnecessary, just serializing `Identifier` would be enough. Also, introducing operators as aliases for functions like `>*<` and `>|<` seems more confusing than helpful. Could you boil this down more, please?

Also, don't you think the test would be easier to understand if it used something like `Hashable` as suggested here: https://github.com/soenkehahn/generics-eot/issues/6#issuecomment-370201049? I realize it's taking me some time to wrap my head around `Divisable` and `Decidable` and test-cases are meant to be good documentation. So if we can avoid unnecessarily complex examples, we should do that, I think.

I would also perfectly understand if you wouldn't be too excited about working on a PR with `Hashable`, since it's not your use-case. If that's the case, please, let me know and we'll try to figure something else out.